File: langs/python3/problems/gpu_scheduler/stub.py
class Simulation:
    def __init__(self):
        pass

    def add_gpu(self, gpu_id, mem):
        """Register a device. "true", "false", or "invalid_request"."""
        raise NotImplementedError

    def submit_job(self, job_id, mem):
        """Queue a job. "true", "false", or "invalid_request"."""
        raise NotImplementedError

    def status(self, job_id):
        """queued, running, done, or ""."""
        raise NotImplementedError

    def assign(self):
        """Start one queued job. Job id or ""."""
        raise NotImplementedError

    def complete(self, job_id):
        """Finish a running job. "true" or "invalid_request"."""
        raise NotImplementedError

    def cancel(self, job_id):
        """Drop a queued or running job. "true" or "invalid_request"."""
        raise NotImplementedError

    def set_priority(self, job_id, priority):
        """Set a queued job's rank. "true" or "invalid_request"."""
        raise NotImplementedError

File: langs/java/problems/gpu_scheduler/stub.java
public class Simulation {
    public Simulation() {}

    /**
     * Register a device. {@code "true"}, {@code "false"}, or
     * {@code invalid_request}.
     */
    public String addGpu(String gpuId, int mem) {
        return "";
    }

    /**
     * Queue a job. {@code "true"}, {@code "false"}, or
     * {@code invalid_request}.
     */
    public String submitJob(String jobId, int mem) {
        return "";
    }

    /**
     * queued, running, done, or {@code ""}.
     */
    public String status(String jobId) {
        return "";
    }

    /**
     * Start one queued job. Job id or {@code ""}.
     */
    public String assign() {
        return "";
    }

    /**
     * Finish a running job. {@code "true"} or {@code invalid_request}.
     */
    public String complete(String jobId) {
        return "";
    }

    /**
     * Drop a queued or running job. {@code "true"} or
     * {@code invalid_request}.
     */
    public String cancel(String jobId) {
        return "";
    }

    /**
     * Set a queued job's rank. {@code "true"} or {@code invalid_request}.
     */
    public String setPriority(String jobId, int priority) {
        return "";
    }
}
